src/calendar/model/CalendarModel.ts

```diff
@@ -255,24 +254,45 @@ export class CalendarModel {
 	}
 
 	private async handleCalendarEventUpdate(update: CalendarEventUpdate): Promise<void> {
+		// we want to delete the CalendarEventUpdate after we are done, even, in some cases, if something went wrong.
+		let erase = true
```

Comment:
I'm not a fan of this pre-initialized variable. I think we should either explicitly set it in each case (there are not so many of them) or better yet invoke some `this.eraseUpdate()` in each case.